### PR TITLE
(SHOULD PROBABLY BE TM ONLY) (PORT) Ports Armstrong Style.

### PR DESCRIPTION
**Pull request by Yawet330**
## About The Pull Request
Ports armstrong style from hippiestation; new moves.
The martial art works off of getting stronger and stronger based off of how long and how frequently you have been fighting people.

## Why It's Good For The Game
Code improvement, new martial arts, variation, new code to support it. Relatively well balanced with no range deflect on a melee only martial arts. Sec can and will delete this if need be.

## Changelog
:cl:
code: literally makes it so help intent is checked for martial arts and other actions... come on guys.
refactor: refactors martial arts slightly to support the new intent.
add: Armstrong style from hippiestation (Credits to the original coders!)
imageadd: hippiestation icons
soundadd: hippiestation sounds.
/:cl:

### CODE REVIEW

**Comment by BobJoga** [comment]
POV: victim of armstrong style in bobmed
image

**Comment by Yawet330** [comment]
image

**Comment by Yawet330** [comment]
needs a speedmerge imo

**Comment by Echopsychotic** [comment]
What is armstrong style?

**Comment by calibraptor** [comment]
A martial art based on how much you've been murderboning doesn't sounds like a good fit for skyrat imho.

**Comment by Yawet330** [comment]
> A martial art based on how much you've been murderboning doesn't sound like a good fit for skyrat imho.

Murderboning will get you instantly destroyed because it has zero ranged protection or damage resists when compared to other martial arts. Literally security are designed and 100% encouraged to just disabler you to the ground. It is a MELEE ONLY art with zero ranged abilities until level 30(the max) which can take over ~30 minutes to get to of CONTINUOUS fighting. Also you say this but we still have motherbase

**Comment by calibraptor** [comment]
> 
> 
> > A martial art based on how much you've been murderboning doesn't sound like a good fit for skyrat imho.
> 
> Murderboning will get you instantly destroyed because it has zero ranged protection or damage resists when compared to other martial arts. Literally security are designed and 100% encouraged to just disabler you to the ground. It is a MELEE ONLY art with zero ranged abilities until level 30(the max) which can take over ~30 minutes to get to of CONTINUOUS fighting. Also you say this but we still have motherbase

This doesn't sounds like an argument in favor of it. If it's underwhelming and gimped by current escalation rules and our level of RP then why add it in?

**Comment by Yawet330** [comment]
> > > A martial art based on how much you've been murderboning doesn't sound like a good fit for skyrat imho.
> > 
> > 
> > Murderboning will get you instantly destroyed because it has zero ranged protection or damage resists when compared to other martial arts. Literally security are designed and 100% encouraged to just disabler you to the ground. It is a MELEE ONLY art with zero ranged abilities until level 30(the max) which can take over ~30 minutes to get to of CONTINUOUS fighting. Also you say this but we still have motherbase
> 
> This doesn't sound like an argument in favor of it. If it's underwhelming and gimped by current escalation rules and our level of RP then why add it in?

The same reasons we covered in main-dev? Its a gimmick item that involves having to setup a gladiator-esque environment to avoid death, its also a well balanced and thought out item that can allow for RP (moreso than CQC, carp, and RB)

**Comment by calibraptor** [comment]
> 
> 
> > > > A martial art based on how much you've been murderboning doesn't sound like a good fit for skyrat imho.
> > > 
> > > 
> > > Murderboning will get you instantly destroyed because it has zero ranged protection or damage resists when compared to other martial arts. Literally security are designed and 100% encouraged to just disabler you to the ground. It is a MELEE ONLY art with zero ranged abilities until level 30(the max) which can take over ~30 minutes to get to of CONTINUOUS fighting. Also you say this but we still have motherbase
> > 
> > 
> > This doesn't sound like an argument in favor of it. If it's underwhelming and gimped by current escalation rules and our level of RP then why add it in?
> 
> The same reasons we covered in main-dev? Its a gimmick item that involves having to setup a gladiator-esque environment to avoid death, its also a well balanced and thought out item that can allow for RP (moreso than CQC, carp, and RB)

Other martial arts stand up on their own merits without requiring you do to a very specific gimmick in order to not get dunked by security.

**Comment by Toxici11i** [comment]
how the fuck is TGstation.dme conflicting lmaaooo

**Comment by Yawet330** [comment]
image
how

**Comment by Yawet330** [comment]
recommending that this is a testmerge-only PR for now while various features are re-balanced and redesigned as a martial art.

File: code/datums/martial/_martial.dm [hunk 1]
```diff
@@ -19,6 +19,9 @@
 /datum/martial_art/proc/disarm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
 	return FALSE
 
+/datum/martial_art/proc/help_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	return FALSE
```
Comment (jjpark-kb) [inline code/datums/martial/_martial.dm:23]: not modular. Needs either to be modular or have //skyrat edit

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -1439,6 +1439,9 @@ GLOBAL_LIST_EMPTY(roundstart_race_datums)
 //////////////////
 
 /datum/species/proc/help(mob/living/carbon/human/user, mob/living/carbon/human/target, datum/martial_art/attacker_style)
+	if(attacker_style && attacker_style.help_act(user,target)) // thank you hippiestation, bless.
+		return 1
```
Comment (jjpark-kb) [inline code/modules/mob/living/carbon/human/species.dm:1443]: not modular. Needs either to be modular or have //skyrat edit

File: code/datums/martial/_martial.dm (hunk 1 above)
Comment (Yawet330) [reply code/datums/martial/_martial.dm:23]: will do

File: code/modules/mob/living/carbon/human/species.dm (hunk 2 above)
Comment (Yawet330) [reply code/modules/mob/living/carbon/human/species.dm:1443]: ditto

**Comment by Yawet330** [comment]
okay code should no longer send the server to brazil with bobmed.

**Pull request closed without merging**